Firestore/Example/Tests/Local/FSTLevelDBMigrationsTests.mm

```diff
@@ -361,6 +367,57 @@ - (void)testRemovesMutationBatches {
   }
 }
 
+- (void)testSetsLastAcknowledgedBatchIdToUnknown {
+  LevelDbMigrations::RunMigrations(_db.get(), 5);
+
+  {
+    LevelDbTransaction transaction(_db.get(), "Setup users");
+
+    FSTPBMutationQueue *fooQueue = [[FSTPBMutationQueue alloc] init];
+    fooQueue.lastAcknowledgedBatchId = 1;
+    transaction.Put(LevelDbMutationQueueKey::Key("foo1"), fooQueue);
+
+    FSTPBMutationQueue *barQueue = [[FSTPBMutationQueue alloc] init];
+    barQueue.lastAcknowledgedBatchId = 2;
+    transaction.Put(LevelDbMutationQueueKey::Key("foo2"), barQueue);
+
+    FSTPBMutationQueue *bazQueue = [[FSTPBMutationQueue alloc] init];
+    bazQueue.lastAcknowledgedBatchId = -1;
+    transaction.Put(LevelDbMutationQueueKey::Key("foo3"), bazQueue);
+
+    transaction.Commit();
+  }
+
+  auto get_last_acked = [&] {
```

Comment:
This mechanism is not cheaply portable to Android. Consider passing the mutation_queue as a parameter to a regular function.